packages/styleguide/src/components/ColorLabel/index.tsx

```diff
@@ -0,0 +1,63 @@
+import { css } from 'glamor'
+import React, { ReactNode } from 'react'
+
+// TODO: these should be out of the SG and usable by story components
+export const COLOR_LABEL_EVENTS = {
+  enter: 'enterColorLabel',
+  leave: 'leaveColorLabel',
+}
+
+export const ColorLabel = ({
+  color,
+  children,
+}: {
+  color: string
+  children: ReactNode
+}) => {
+  const enterEvent = new CustomEvent(COLOR_LABEL_EVENTS.enter, {
+    bubbles: true,
+    detail: { color },
+  })
+  const leaveEvent = new CustomEvent(COLOR_LABEL_EVENTS.leave, {
+    bubbles: true,
+    detail: { color },
+  })
+
+  return (
+    <span
+      {...styles.container}
+      onMouseEnter={() => document.dispatchEvent(enterEvent)}
+      onMouseLeave={() => document.dispatchEvent(leaveEvent)}
```

Comment:
I am not completely sure if it's a good idea to dispatch custom events on the document (but maybe it's fine). Anyway, we need to think about how this will/should work on mobile.

Reply:
yeah the dom events are also a source of concerns for me – for now i just went with the make-it-work approach… if it proves brittle upon testing, i guess we could replace those with some kind of react context.

Reply:
just tested: on staging and on mobile, seems to be working ok… the app is uncharted territory though.